### PR TITLE
Button Group component

### PR DESCRIPTION
**Pull request by shaswat-indian**
This PR creates the `ButtonGroup` component.

https://github.com/reaviz/reablocks/assets/33908100/4d33f505-bfd0-41a0-8e14-3825778b2ffd

Resolves https://github.com/reaviz/reablocks/issues/17

### CODE REVIEW

File: src/elements/Button/Button.tsx [hunk 1]
```diff
@@ -76,38 +77,46 @@ export const Button: FC<ButtonProps & ButtonRef> = forwardRef(
...
+        className={classNames(
+          css.btn,
+          {
+            [css.fullWidth]: fullWidth,
+            [css.disableMargins]: disableMargins,
+            [css.disablePadding]: disablePadding,
+            [css[color]]: true,
+            [css[groupSize || size]]: true,
+            [css[groupVariant || variant]]: true,
+            [css['group']]: !!groupVariant && !!groupSize
```
Comment (amcdnl) [inline src/elements/Button/Button.tsx:99]:
```suggestion
            [css.group]: !!groupVariant && !!groupSize
```

File: src/elements/Button/ButtonGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React, { FC, useMemo } from 'react';
+import {
+  ButtonGroupContext,
+  ButtonGroupContextProps
+} from './ButtonGroupContext';
+
+export interface ButtonGroupProps extends ButtonGroupContextProps {
```
Comment (amcdnl) [inline src/elements/Button/ButtonGroup.tsx:7]: I would add the `onChange` and the `selected` value probably. How does MUI / others do this?

**Review by amcdnl — CHANGES_REQUESTED** [review]
Minor feedback on selections

File: src/elements/Button/ButtonGroup.tsx (hunk 2 above)
Comment (shaswat-indian) [reply src/elements/Button/ButtonGroup.tsx:7]: [MUI](https://mui.com/material-ui/api/button-group/#props) and others don't have these props and I think it makes sense too as unlike a `RadioGroup`, a `ButtonGroup` just groups the buttons into a unified display so it does not need to have a `selected` value or a `onChange` prop.

**Review by amcdnl — APPROVED** [review]

**Pull request merged**